app/tests/integration/nimbus/test_remote_settings.py

```diff
@@ -90,3 +94,49 @@ def test_end_experiment_and_reject_end(
     Dashboard(selenium, kinto_review_url).open().reject()
 
     SummaryPage(selenium, experiment_url).open().wait_for_rejected_alert()
+
+
+@pytest.mark.run_once
+def test_create_new_experiment_approve_check_targeting(
+    selenium,
+    base_url,
+    kinto_url,
+    kinto_review_url,
+    experiment_url,
+    default_data,
+    create_experiment,
+    slugify,
+):
+    default_data.audience.targeting = (
+        BaseExperimentAudienceTargetingOptions.TARGETING_MAC_ONLY
```

Comment:
So it looks like this test is using only a single targeting config option, but the ticket says

"So ideally we'd have a test for each targeting config that ... Creates a Nimbus experiment for Desktop with that targeting config and no other custom targeting (we might be able to do this from the command line without needing to start the dev server at all)"

So we'll need to find a way to capture every targeting config that is landed in experimenter and do a test for each one.  Ideally we'll be able to discover all the targeting configs that are defined in NimbusConstants rather than hard coding them here so that when someone adds a new one they don't also have to come modify the integration tests.

Reply:
Couple ways I can think of to do it are

1) Discover the list from the drop down options in the UI, but that would depend on creating an experiment first to get to that page then pulling all the options out and then creating an experiment for each one

2) Use a management command to create an experiment with each targeting config during the setup phase in the Makefile which can then just import the TargetingConfig list directly and create them with predictable names that the tests can then discover

3) Query the configuration endpoint with the graphql API which will return them all as JSON

Reply:
"capture every targeting config", where would this be defined?

Reply:
This is probably the easiest way to do it, call the V5 graphql API and it'll return them all:

```python
In [14]: requests.post('https://localhost/api/v5/graphql', json={'operationName': 'getConfig',
    ...:  'variables': {},
    ...:  'query': '\nquery getConfig {\n  nimbusConfig {\n    targetingConfigs {\n      label\n      value\n      applicationValues\n    }\n  }\n}\n'}, verify=Fals
    ...: e).content
/Users/jaredlockhart/Library/Caches/pypoetry/virtualenvs/app-9BVIRsKI-py3.8/lib/python3.8/site-packages/urllib3/connectionpool.py:981: InsecureRequestWarning: Unverified HTTPS request is being made to host 'localhost'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
  warnings.warn(
Out[14]: b'{"data":{"nimbusConfig":{"targetingConfigs":[{"label":"No Targeting","value":"","applicationValues":["DESKTOP","FENIX","IOS","FOCUS_ANDROID","KLAR_ANDROID"]},{"label":"First start-up users","value":"first_run","applicationValues":["DESKTOP"]},{"label":"First start-up users from Chrome","value":"first_run_chrome","applicationValues":["DESKTOP"]},{"label":"First start-up users on Windows 10 1903 (build 18362) or newer","value":"first_run_win1903","applicationValues":["DESKTOP"]},{"label":"Homepage set to google.com","value":"homepage_google_dot_com","applicationValues":["DESKTOP"]},{"label":"Urlbar (Firefox Suggest)","value":"urlbar_firefox_suggest","applicationValues":["DESKTOP"]},{"label":"Mac OS users only","value":"mac_only","applicationValues":["DESKTOP"]},{"label":"No enterprise users","value":"no_enterprise_users","applicationValues":["DESKTOP"]},{"label":"First start-up users","value":"first_run_mobile","applicationValues":["FENIX","IOS"]},{"label":"Infrequent user (uris)","value":"infrequent_user_uris","applicationValues":["DESKTOP"]},{"label":"Casual user (uris)","value":"casual_user_uris","applicationValues":["DESKTOP"]},{"label":"Regular user (uris)","value":"regular_user_uris","applicationValues":["DESKTOP"]},{"label":"Core user (uris)","value":"core_user_uris","applicationValues":["DESKTOP"]},{"label":"No enterprise or past VPN use","value":"no_enterprise_or_past_vpn","applicationValues":["DESKTOP"]}]}}}'
```